waku/node/v2/waku_types.nim

```diff
@@ -154,3 +174,12 @@ proc generateRequestId*(rng: ref BrHmacDrbgContext): string =
   var bytes: array[10, byte]
   brHmacDrbgGenerate(rng[], bytes)
   toHex(bytes)
+
+proc computeIndex* (w: WakuMessage): Index=
+  ## Takes a WakuMessage and returns its index
+  let payloadStr=w.payload.join("") #convers payload to string
+  var data: string
+  data.add(w.contentTopic)
+  data.add(payloadStr)
+  result.digest= $secureHash(data) #computes sha1 of data
+  result.timeStamp= $getTime().utc #gets the timestamp in utc with seconds precision
```

Comment:
```suggestion
proc computeIndex*(w: WakuMessage): Index =
  ## Takes a WakuMessage and returns its index
  let payloadStr = w.payload.join("") # converts payload to string
  var data: string
  data.add(w.contentTopic)
  data.add(payloadStr)
  result.digest = $secureHash(data) # computes sha1 of data
  result.timeStamp = $getTime().utc # gets the timestamp in utc with seconds precision
```

Reply:
you can also use `&` for string concatenation, so it could probably be written something like (untested, assuming $ is implemented on payload):

let data = msg.contentTopic & $msg.payload

Reply:
Few more comments:

- I'd make it clear that the timestamp is for receiver node, so `receivedTime` or something like this.

- I believe contentTopic is optional, so might want to make sure this is handled.

- Perhaps a test here would be useful to make sure it works as intended?

Reply:
> Perhaps a test here would be useful to make sure it works as intended?

I was going to implement the tests in the next issue, but per your comment, I'll include it in this PR.

Reply:
> I'd make it clear that the timestamp is for receiver node, so receivedTime or something like this.
> 
> I believe contentTopic is optional, so might want to make sure this is handled.

Thanks, addressed!